src/Symfony/Component/Uid/Command/GenerateUuidCommand.php

```diff
@@ -0,0 +1,194 @@
+                new InputOption('random-based', null, InputOption::VALUE_NONE, 'To generate a random based UUID'),
+                new InputOption('count', 'c', InputOption::VALUE_REQUIRED, 'The number of UUID to generate', 1),
+                new InputOption('format', 'f', InputOption::VALUE_REQUIRED, 'The UUID output format: canonical, base_58 or base_32', 'canonical'),
+            ])
+            ->setDescription(self::$defaultDescription)
+            ->setHelp(<<<'EOF'
+The <info>%command.name%</info> generates a UUID.
+
+    <info>php %command.full_name%</info>
+
+To generate a time based UUID:
+
+    <info>php %command.full_name% --time-based=now</info>
+
+To specify a time based UUID's node:
+
+    <info>php %command.full_name% --time-based=@1613480254 --node=fb3502dc-137e-4849-8886-ac90d07f64a7</info>
+
+To generate a name based UUID:
+
+    <info>php %command.full_name% --name-based=foo</info>
+
+To specify a name based UUID's namespace:
+
+    <info>php %command.full_name% --name-based=bar --namespace=fb3502dc-137e-4849-8886-ac90d07f64a7</info>
+
+To generate a random based UUID:
+
+    <info>php %command.full_name% --random-based</info>
+
+To generate several UUIDs:
+
+    <info>php %command.full_name% --count=10</info>
+
+To output a specific format:
+
+    <info>php %command.full_name% --format=base_58</info>
+EOF
+            )
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output instanceof ConsoleOutputInterface ? $output->getErrorOutput() : $output);
+
+        $time = $input->getOption('time-based');
+        $node = $input->getOption('node');
+        $name = $input->getOption('name-based');
+        $namespace = $input->getOption('namespace');
+        $random = $input->getOption('random-based');
+
+        switch (true) {
+            case !$time && !$node && !$name && !$namespace && !$random:
+                $create = [$this->factory, 'create'];
+
+                break;
+            case $time && !$name && !$namespace && !$random:
+                if ($node) {
+                    try {
+                        $node = Uuid::fromString($node);
+                    } catch (\InvalidArgumentException $e) {
+                        $io->error(sprintf('Invalid node "%s". It cannot be parsed.', $node));
+
+                        return 1;
+                    }
+                }
+
+                try {
+                    $time = new \DateTimeImmutable($time);
+                } catch (\Exception $e) {
+                    $io->error(sprintf('Invalid timestamp "%s". It cannot be parsed.', $input->getOption('time-based')));
+
+                    return 2;
+                }
+
+                if ($time < new \DateTimeImmutable('@-12219292800')) {
```

Comment:
we don't have to check for these, the implementations do it already

Reply:
I thought it would be better to check it to display a better error message. It is better than an uncaught exception + "The given UUID date cannot be earlier than 1582-10-15" IMO.

Reply:
We can catch the logic exception and use $e->getMessage() instead.
We should not duplicate such logic in commands IMHO